ext/mysqlnd/mysqlnd_auth.c

```diff
@@ -694,20 +694,149 @@ mysqlnd_xor_string(char * dst, const size_t dst_len, const char * xor_str, const
 	}
 }
 
+#ifndef PHP_WIN32
 
 #include <openssl/rsa.h>
 #include <openssl/pem.h>
 #include <openssl/err.h>
 
+typedef RSA * mysqlnd_rsa_t;
+
+/* {{{ mysqlnd_sha256_get_rsa_from_pem */
+static mysqlnd_rsa_t
+mysqlnd_sha256_get_rsa_from_pem(const char *buf, size_t len)
+{
+	RSA * ret = NULL;
+	BIO * bio = BIO_new_mem_buf(buf, len);
+	ret = PEM_read_bio_RSA_PUBKEY(bio, NULL, NULL, NULL);
```

Comment:
You can just write `RSA *ret = ...` here and save the declaration above.

Reply:
It seems to me that we need a "portable" type declaration for some common code, e.g. https://github.com/php/php-src/blob/797a3fcc2ac8226bb0dd0eaf647b6f66d3c803c6/ext/mysqlnd/mysqlnd_auth.c#L839

Reply:
Sorry, what I meant is just the `RSA * ret = NULL` line :)

Reply:
Ah, of course!